src/main/kotlin/ecommerce/stripe/StripeClient.kt

```diff
@@ -0,0 +1,91 @@
+package ecommerce.stripe
+
+import com.fasterxml.jackson.module.kotlin.jacksonObjectMapper
+import ecommerce.dto.PaymentErrorResponse
+import ecommerce.dto.PaymentRequest
+import ecommerce.dto.PaymentResponse
+import org.springframework.http.HttpHeaders
+import org.springframework.http.MediaType
+import org.springframework.stereotype.Component
+import org.springframework.web.client.RestClient
+import org.springframework.web.client.RestClientResponseException
+import kotlin.collections.joinToString
+import kotlin.jvm.java
+
+@Component
+class StripeClient(
+    private val stripeProperties: StripeProperties,
+) {
+    private val restClient: RestClient = RestClient.create()
+
+    fun createCheckoutSession(req: PaymentRequest): PaymentResponse {
+        val body =
+            listOf(
+                "amount=${req.amount}",
+                "currency=${req.currency}",
+                "payment_method=${req.paymentMethod}",
+                "confirm=true",
+                "automatic_payment_methods[enabled]=true",
+                "automatic_payment_methods[allow_redirects]=never",
+            ).joinToString("&")
+
+        return try {
+            val response =
+                restClient.post()
+                    .uri("https://api.stripe.com/v1/payment_intents")
+                    .header(HttpHeaders.AUTHORIZATION, "Bearer ${stripeProperties.secretKey}")
+                    .contentType(MediaType.APPLICATION_FORM_URLENCODED)
+                    .body(body)
+                    .retrieve()
+                    .toEntity(PaymentResponse::class.java)
+
+            response.body ?: throw IllegalStateException("Empty response from Stripe")
+        } catch (e: RestClientResponseException) {
+            val errorJson = e.responseBodyAsString
+            val stripeError = parseStripeError(errorJson)
+            PaymentResponse(
+                id = "",
+                amount = req.amount,
+                status = "FAILED",
+                clientSecret = null,
+                errorMessage = convertMessageFromError(stripeError),
+            )
+        } catch (e: Exception) {
+            throw IllegalArgumentException("Stripe error: ${e.message}", e)
+        }
+    }
+
+    private fun parseStripeError(json: String?): PaymentErrorResponse? {
+        if (json.isNullOrBlank()) return null
+        return try {
+            val root = jacksonObjectMapper().readTree(json)
```

Comment:
We can also inject `ObjectMapper` dependency through constructor and use it like a singleton.